frb_example/pure_dart/dart/lib/main.dart

```diff
@@ -538,11 +537,10 @@ void main(List<String> args) async {
     expect(await SomeStruct(bridge: api, value: 6).nonStaticReturnOkCustomError(), 6);
   });
 
-/*
   test('Throw anyhow error', () async {
-    expect(() async => await api.throwAnyhow(), throwsA("anyhow error"));
+    expect(() async => await api.throwAnyhow(), throwsA(isA<FrbException>()));
   });
-*/
+
```

Comment:
Btw we may need to add a test case: A rust function which returns *custom errors*, does *do panic*, and Dart side should get correct results.

e.g.

```rust
fn f() -> Result<Something, CustomError> {
  panic!(....)
}
```

```dart
expect(() => api.f(...), throwsA(isA<PanicError>()));
```

Reply:
This is already done in

```dart
  test('dart call returnPanic', () async {
    try {
      await api.returnPanic();
      fail("exception not thrown");
    } catch (e) {
      print('dart catch e: $e');
      expect(e, isA<FrbException>());
    }
  });
```

Reply:
I guess `returnPanic` function signature returns a `anyhow::Result`, not a `Result<..., SomeCustomErrorType>`? Thus the new test is needed.

Reply:
Btw the full list is as follows:

- [ ] rust function signature has no Result (e.g. `adder(...) -> i32`) + it panics (Do we have existing tests?)
- [x] rust function signature has anyhow::Result + it returns error (should have existing tests)
- [x] rust function signature has anyhow::Result + it panics (should have existing tests)
- [x] rust function signature has Result<..., OurCustomErrorTypes> + it returns error (seems we have added tests in this pr already)
- [ ] rust function signature has Result<..., OurCustomErrorTypes> + it panics

Reply:
first one is 

```rust
pub fn return_panic() -> i32 {
    panic!("return_panic() is called, thus deliberately panic")
}
```

which is already being tested: 

```dart
  test('dart call returnPanic', () async {
    try {
      await api.returnPanic();
      fail("exception not thrown");
    } catch (e) {
      print('dart catch e: $e');
      expect(e, isA<FrbException>());
    }
  });
```

Reply:
Indeed, the last one fails:

```rust
pub fn panic_with_custom_result() -> Result<(), CustomError> {
    panic!("just a panic");
}
```
because there's no panic catcher:

```dart
  Future<void> panicWithCustomResult({dynamic hint}) => executeNormal(FlutterRustBridgeTask(
        callFfi: (port_) => inner.wire_panic_with_custom_result(port_),
        parseSuccessData: _wire2api_unit,
        parseErrorData: _wire2api_custom_error,
        constMeta: kPanicWithCustomResultConstMeta,
        argValues: [],
        hint: hint,
      ));

```

it expects a custom error.

Any ideas?

Reply:
Maybe we should separate "error" into two categories - "panic error" and "normal errors (none when returning i32, anyhow error when returning anyhow::Result, custom error when returning custom error). Then, `parseErrorData` is renamed to `parseResultErrorData` (or sth like that), while the actual "parsePanicErrorData" is hardcoded and never changed.